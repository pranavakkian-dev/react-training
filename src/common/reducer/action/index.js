import {
  LOGIN, RESET,
} from '../../constants/reducerTypes';

export const handleLogin = (username) => ({
  type: LOGIN,
  payload: username,
});

export const handleReset = () => ({
  type: RESET,
});
