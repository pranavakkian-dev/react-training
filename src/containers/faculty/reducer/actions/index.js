import { FACULTY_RESET, ADD_FACULTY, DELETE_FACULTY } from '../../../../common/constants/reducerTypes';

export const handleAddFaculty = (faculty) => ({
  type: ADD_FACULTY,
  payload: faculty,
});

export const handleDeleteFaculty = (faculty) => ({
  type: DELETE_FACULTY,
  payload: faculty,
});

export const handleFacultyReset = () => ({
  type: FACULTY_RESET,
});
