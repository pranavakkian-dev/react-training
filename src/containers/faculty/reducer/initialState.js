const initialState = {
  faculties: [],
};

export default initialState;
