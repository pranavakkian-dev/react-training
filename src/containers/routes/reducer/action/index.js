import {
  LOGIN, RESET,
} from '../../../../common/constants/reducerTypes';

export const handleLogin = (username) => ({
  type: LOGIN,
  payload: username,
});

export const handleReset = () => ({
  type: RESET,
});
