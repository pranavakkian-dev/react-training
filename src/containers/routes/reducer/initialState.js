const initialState = {
  loggedIn: false,
  userName: null,
};

export default initialState;
