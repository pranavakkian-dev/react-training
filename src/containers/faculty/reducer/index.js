import initialState from './initialState';
import { FACULTY_RESET, ADD_FACULTY, DELETE_FACULTY } from '../../../common/constants/reducerTypes';

const reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case ADD_FACULTY: {
      const { faculties } = state;
      return { ...state, faculties: [...faculties, { ...payload }] };
    }
    case DELETE_FACULTY: {
      const { faculties } = state;
      const updatedFaculties = faculties.filter((faculty) => faculty !== payload);
      return { ...state, faculties: [...updatedFaculties] };
    }
    case FACULTY_RESET:
      return initialState;
    default:
      return state;
  }
};

export default reducer;
