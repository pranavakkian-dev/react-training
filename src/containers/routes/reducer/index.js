import initialState from './initialState';
import {
  LOGIN, RESET,
} from '../../../common/constants/reducerTypes';

const reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case LOGIN:
      return { ...state, loggedIn: true, userName: payload };
    case RESET:
      return initialState;
    default:
      return state;
  }
};

export default reducer;
