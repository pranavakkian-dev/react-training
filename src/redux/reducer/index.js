import initialState from './initialState';
import {
  LOGIN, RESET,
} from '../../common/constants/reducerTypes';

const Reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case LOGIN:
      return { ...state, loggedIn: true, userName: payload };
    case RESET:
      return initialState;
    default:
      return state;
  }
};

export default Reducer;
