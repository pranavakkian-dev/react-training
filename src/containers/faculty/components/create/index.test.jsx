import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Create from './index';
import { handleAddFaculty } from '../../reducer/actions';

describe('Create', () => {
  const mockedDispatch = jest.fn();

  test('Component rendered successfully!', () => {
    const { container } = render(<Create dispatch={mockedDispatch} />);
    expect(container).toMatchSnapshot();
  });

  test('Heading rendered successfully!', () => {
    render(<Create dispatch={mockedDispatch} />);
    expect(screen.getByRole('heading', { name: 'Add Faculty' })).toBeInTheDocument();
  });

  test('ID input captured successfully!', () => {
    render(<Create dispatch={mockedDispatch} />);
    expect(screen.getByRole('spinbutton', { name: 'ID:' })).toHaveValue(null);
    userEvent.type(screen.getByRole('spinbutton', { name: 'ID:' }), '100');
    expect(screen.getByRole('spinbutton', { name: 'ID:' })).toHaveValue(100);
  });

  test('Name input captured successfully!', () => {
    render(<Create dispatch={mockedDispatch} />);
    expect(screen.getByRole('textbox', { name: 'Name:' })).toHaveValue('');
    userEvent.type(screen.getByRole('textbox', { name: 'Name:' }), 'user');
    expect(screen.getByRole('textbox', { name: 'Name:' })).toHaveValue('user');
  });

  test('Faculty created successfully!', () => {
    render(<Create dispatch={mockedDispatch} />);
    userEvent.type(screen.getByRole('spinbutton', { name: 'ID:' }), '100');
    userEvent.type(screen.getByRole('textbox', { name: 'Name:' }), 'user');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    expect(mockedDispatch).toHaveBeenNthCalledWith(1, handleAddFaculty({ id: '100', name: 'user' }));
    expect(screen.getByRole('spinbutton', { name: 'ID:' })).toHaveValue(null);
    expect(screen.getByRole('textbox', { name: 'Name:' })).toHaveValue('');
  });

  test('Faculty creation un-successful!', () => {
    render(<Create dispatch={mockedDispatch} />);
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    expect(mockedDispatch).not.toHaveBeenCalled();
  });
});
