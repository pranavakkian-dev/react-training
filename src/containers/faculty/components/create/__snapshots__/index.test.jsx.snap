// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Component rendered successfully! 1`] = `
<div>
  <h4>
    Add Faculty
  </h4>
  <label
    for="facultyId"
  >
    ID:
    <br />
    <input
      id="facultyId"
      type="number"
      value=""
    />
  </label>
  <br />
  <label
    for="facultyName"
  >
    Name:
    <br />
    <input
      id="facultyName"
      value=""
    />
  </label>
  <br />
  <button
    type="button"
  >
    Submit
  </button>
</div>
`;
